src/fsharp/pars.fsy

```diff
@@ -1155,7 +1155,10 @@ fileModuleImpl:
       (fun (isRec, path, xml) -> 
         match path with 
         | [] -> ParsedImplFileFragment.AnonModule($1, m)  
-        | _ -> ParsedImplFileFragment.NamespaceFragment(path, isRec, SynModuleOrNamespaceKind.DeclaredNamespace, $1, xml, [], m)) } 
+        | _ ->
+            let lastDeclRange = List.tryLast $1 |> Option.map (fun decl -> decl.Range) |> Option.defaultValue (rhs parseState 1)
+            let m = mkRange lastDeclRange.FileName (lhs parseState).Start lastDeclRange.End
```

Comment:
Maybe, `unionRanges`?

Reply:
unionRanges did not work as `lhs parseState` captured a range to was too large.
In case of